### PR TITLE
Decks state in Home function and handleDeckSubmit

### PR DESCRIPTION
**Pull request by nleech16**
Probably not perfect but want some feedback! I considered taking out the Decks component completely but wasn't sure if we had other uses for it.

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -24,16 +43,15 @@ export default function Home() {
 
   class Decks extends React.Component<DeckProps, DeckState> {
     render() {
-      const listItems = this.props.decks.map((deck) => (
-        <Link href={`/review/${deck.id}`}>
+      return this.props.decks.map((deck) => (
+        <Link href={`/review/${deck.id}`} key={deck.id}>
```
Comment (nleech16) [inline src/pages/index.tsx:47]: console was complaining when there was no key

Comment (SlyPuffin) [reply src/pages/index.tsx:47]: Nice! Did you notice this when you ran `npm run dev`? I don't pay enough attention to the warnings and such, but running `npm run lint` gave me that as an error, too. Thanks!

File: src/pages/index.tsx [hunk 2]
```diff
@@ -24,16 +43,15 @@ export default function Home() {
 
   class Decks extends React.Component<DeckProps, DeckState> {
     render() {
-      const listItems = this.props.decks.map((deck) => (
-        <Link href={`/review/${deck.id}`}>
+      return this.props.decks.map((deck) => (
```
Comment (SlyPuffin) [inline src/pages/index.tsx:46]: Cool that you can return the value from `decks.map` and it fits right into the list below.

**Comment by SlyPuffin** [comment]
This was a great opportunity to look into react hooks for me, too 😊
Looking through the docks I found the lint plug-in they officially provided. Wanna add it to this PR?
https://www.npmjs.com/package/eslint-plugin-react-hooks

File: src/pages/index.tsx [hunk 3]
```diff
@@ -16,6 +16,25 @@ export default function Home() {
       refetchOnWindowFocus: false,
     }
   );
+  const addDeckMutation = trpc.useMutation('add-deck');
+
+  const [decks, setDecks] = React.useState<Deck[]>(data?.decks || []);
```
Comment (SlyPuffin) [inline src/pages/index.tsx:21]: I'm still kinda hazy on trpc's useQuery and data vs. isLoading in effect. useState should declare an initial value, which I would guess is empty on first render? Does the "initial state" of the hook change when data loads in or is it something where we should run setDecks on load?

File: src/pages/index.tsx [hunk 4]
```diff
@@ -16,6 +16,25 @@ export default function Home() {
...
+    const name = (e as any).target.elements.name;
+
+    const input = {
+      name: name.value,
+      type: "default",
+      userId: userQuery.data?.user?.id
+    };
+    const deck = await addDeckMutation.mutateAsync(input);
+    decks.push(deck);
+    setDecks([...decks]);
```
Comment (SlyPuffin) [inline src/pages/index.tsx:35]: If you wanted to be trickier, I think you could change these two lines to one (according to the documentation):
`setDecks(prevDeck => prevDeck.push(deck))`

Not that I think that's necessary--just something kinda fancy from the documentation: https://reactjs.org/docs/hooks-reference.html#usestate

**Comment by SlyPuffin** [comment]
> Probably not perfect but want some feedback! I considered taking out the Decks component completely but wasn't sure if we had other uses for it.

In its current state, I agree that the Decks component is probably unnecessary. It might be nice to put the new deck logic and the deck list logic in a component for separation from headers and other buttons later on. That being said, Thomas' implementation of the nav bar is outside of the individual pages, so it wouldn't be relevant in that case. Thoughts?

**Pull request merged**